### PR TITLE
Use anomaly datamodule in CLI

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
### Summary

Addresses: https://github.com/openvinotoolkit/training_extensions/issues/3219

```bash
otx train --config src/otx/recipe/anomaly_classification/padim.yaml --data_root ~/datasets/MVTec/bottle --work_dir otx-workspace
otx test --config src/otx/recipe/anomaly_classification/padim.yaml --data_root ~/datasets/MVTec/bottle --work_dir otx-workspace 
```

### Padim
image

### Stfpm
image

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/recipe/anomaly_classification/draem.yaml [hunk 1]
```diff
@@ -31,11 +32,10 @@ overrides:
         batch_size: 8
         num_workers: 4
         transforms:
-          - class_path: otx.core.data.transform_libs.torchvision.ResizetoLongestEdge
```
Comment (samet-akcay) [inline src/otx/recipe/anomaly_classification/draem.yaml:34]: what is the difference between the native torchvision transform and `otx.core.data.transform_libs.torchvision`. Ideally transforms can be imported directly from `torchvision.transforms.v2.Resize`? @harimkang any thoughts?

@ashwinvaidya17, this way of transform configuration should ideally be in anomalib too

**Review by samet-akcay — DISMISSED** [review]
Thanks for addressing this @ashwinvaidya17. Let's prioritise the discussion of custom datamodule support.

File: src/otx/recipe/anomaly_classification/draem.yaml (hunk 1 above)
Comment (harimkang) [reply src/otx/recipe/anomaly_classification/draem.yaml:34]: Um, looking at it, this looks like a transform for the visual_prompting model, we can use `torchvision.transforms.v2.Resize`

**Comment by harimkang** [comment]
It looks like the test is still not passing, please take a look at it, and maybe we shouldn't put this commit in 2.1 and just leave it as a workaround for 2.0, because I think we should fix it the right way in 2.1. It would be great if you could work on develop when you have time.

**Review by harimkang — APPROVED** [review]

**Review by jaegukhyun — APPROVED** [review]

**Pull request merged**